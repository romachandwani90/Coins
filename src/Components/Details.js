import React from 'react';
import { useEffect, useState } from "react";
import { getDetails } from '../services/service';
import styled from 'styled-components';

const Heading = styled.div`
    display: flex;
    align-items: center;
    margin: 25px 40px;

    span {
        padding: 0 0 0 10px
    }
`;

const Container = styled.div`
    margin: 10px 40px;
    background: #eee;
    padding: 15px 20px;
    border-radius: 6px;
`;

const Details = (props) => {
    const { id } = props.match.params;
    const [details, setDetails] = useState({});

    const isEmptyObject = (value) => {
        return Object.keys(value).length === 0 && value.constructor === Object;
    }
    useEffect(() => {
        const fetchData = async () => {
            const data = await getDetails(id);
            setDetails(data)
        };
        if (isEmptyObject(details)) {
            fetchData();
        }
    });
    return <div>{
        !isEmptyObject(details) && <div>
            <Heading>
                <img src={details.image.small} alt={details.id} />
                <span>{details.name}</span>
            </Heading>
            <Container>
                <div><b>Symbol:</b> {details.symbol}</div>
                <div><b>Hashing algorithm:</b> {details.hashing_algorithm || 'Unknown'}</div>
                <div><b>Description:</b> <div dangerouslySetInnerHTML={{__html: details.description['en']}} /></div>
                <div><b>Market cap:</b> â‚¬{details.market_data.market_cap['eur']}</div>
                
                <div><b>Homepage:</b> {details.links.homepage.map(h => h) }</div>
                <div><b>Genesis Date:</b> {details.genesis_date}</div>
            </Container>
        </div>    
    }</div>
};

export default Details;