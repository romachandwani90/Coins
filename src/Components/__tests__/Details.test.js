import React from 'react'
import {render} from '@testing-library/react'
import Details from '../Details'

describe('Details', () => {
let component;
const props = {
    match: {
        params: {id: 'bitcoin'}
    }
}

const setup = () => {
    component = render(<Details {...props} />);
};


beforeEach(() => {
    setup();
});

it('should take a snapshot', () => {

    expect(component.asFragment()).toMatchSnapshot()
})
});
