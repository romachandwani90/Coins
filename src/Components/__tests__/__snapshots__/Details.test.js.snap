// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details should take a snapshot 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;
